scripts/verify_genproto.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+# This scripts is automatically run by CI to prevent pull requests missing running genproto.sh
+# after changing *.proto file.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+tmpWorkDir=$(mktemp -d -t 'twd.XXXXXX')
+mkdir "$tmpWorkDir/etcd"
+tmpWorkDir="$tmpWorkDir/etcd"
+cp -r . "$tmpWorkDir"
+pushd "$tmpWorkDir"
+git add -A && git commit -m init
+./scripts/genproto.sh
+diff=$(git diff --numstat | awk '{print $3}')
```

Comment:
This script will still compare generated proto versus proto committed, so there is no difference from older script.

Reply:
@serathius 
Sorry for affecting your review. I just force-push a new commit that handles a case that `git commit` will fail.

Reply:
I'm little worried about `git` shenanigans (`add` `commit`), it makes the code very fragile. 

Have you considered avoiding them? I will not push too much as overall change should be beneficial, just worried about over-complicating `verify_genproto.sh` if we could maybe simplify `genproto.sh`

Reply:
I have tried using regular diff. But the genproto.sh will generate some files which are ignored by .gitignore. And as regular diff doesn't support .gitignore, we have to maintain an additional ignore list...

Reply:
Ok, I think we are good enough. We can iterate on the code.